python/ray/ml/utils/gbdt_utils.py

```diff
@@ -0,0 +1,121 @@
+from typing import Dict, Type, Any
+import warnings
+
+from ray import tune
+from ray.util.annotations import DeveloperAPI
+from ray.ml.trainer import Trainer
+from ray.ml.checkpoint import Checkpoint
+from ray.tune import Trainable
+from ray.tune.utils.placement_groups import PlacementGroupFactory
+from ray.ml.constants import PREPROCESSOR_KEY
+
+import xgboost_ray
+
+PARAMS_KEY = "params"
+DMATRIX_PARAMS_KEY = "dmatrix_params"
+
+
+@DeveloperAPI
+class GBDTTrainer(Trainer):
```

Comment:
Would it make sense to have it be its own `Trainer` or as a mixin?

Reply:
+1 I think this would make sense as its own Trainer!

Reply:
The util file is basically just this, I'll simply rename it

Reply:
Yeah that's sufficient! And move it to ml/train directory?

Reply:
Yep